### ISSUE TITLE
Configuration via mocha.opts is deprecated. Use RC files or package.json instead.

### ISSUE THREAD

**Issue by avanderberg** [wix/Detox#2057]
**Is your feature request related to a problem? Please describe.**
When I execute Detox tests I see the console message:
```
detox[88089] ERROR: at internal/process/warning.js:32:3
 (node:88089) DeprecationWarning: Configuration via mocha.opts is DEPRECATED and will be removed from a future version of Mocha. Use RC files or package.json instead.
```

On https://mochajs.org/#mochaopts it says:

> mocha.opts file support is DEPRECATED and will be removed from a future version of Mocha. We recommend using a configuration file instead.

It looks like the Detox configuration options do not yet allow a mocha configuration file because only the `mocha.opts` option is described here:
https://github.com/wix/detox/blob/master/docs/APIRef.Configuration.md#mocha

**Describe the solution you'd like**
I would like to get a solution to use a Mocha configuration file to be able to upgrade Mocha.

**Describe alternatives you've considered**
Maybe it's possible to use Jest instead of Mocha. But Mocha works well for us so far.

**Additional context**
I use the following versions:
- "detox": "^16.4.1",
- "mocha": "^7.1.1",

**Comment by LeoNatan** [comment]
Hello,

Internally, we've moved to mostly using Jest. If you'd like to submit a PR for this feature, we'd appreciate it. Thanks

**Comment by d4vidi** [comment]
Indeed, we strongly recommend using Jest over Mocha, see https://github.com/wix/Detox/blob/master/docs/Guide.Jest.md.
Nevertheless could you elaborate on the context where you're getting this error? In essence, Detox works seamlessly with the underlying test runner of choice. Hence, the error likely comes strictly from mocha itself. Or is there something I missed here?
@noomorph

**Comment by sonicdoe** [comment]
> we strongly recommend using Jest over Mocha

That’s good to know but I’m surprised to hear that. Looking at the [Getting Started](https://github.com/wix/Detox/blob/9b222446214fe913b8528ff78372ec15a30bb342/docs/Introduction.GettingStarted.md#1-install-a-test-runner-running_man) guide, which states both are supported but “Jest is more complex to set up” and “Mocha is easy to set up and is lightweight”, I’d probably lean towards Mocha. Furthermore, even in the [Jest](https://github.com/wix/Detox/blob/9b222446214fe913b8528ff78372ec15a30bb342/docs/Guide.Jest.md) guide, which you’ve linked to, the introduction states that Mocha is the *default* test runner.

Don’t get me wrong, I wouldn’t mind if Detox heavily prefers one test runner over another but the current documentation doesn’t really give me this impression.

**Comment by d4vidi** [comment]
@sonicdoe you're right, sorry if any of the docs misled you. Becoming opinionated about this topic is new to us as well, we will try to keep things up to date.
Nevertheless, that doesn't suggest Mocha shouldn't work. It is maintained and regularly tested. Hence my question - what's the context where you're getting the error?

**Comment by noomorph** [comment]
To get rid of:

```
DeprecationWarning: Configuration via mocha.opts is DEPRECATED and will be removed from a future version of Mocha. Use RC files or package.json instead.
```

1. Upgrade detox to version `16.5.1` or a more recent one.
2. Convert your `mocha.opts` file to `.mocharc` according to https://mochajs.org documentation and point `runner-config` field in `package.json` to that `.mocharc` file:

```json
{
  "detox": {
    "runner-config": "e2e/.mocharc"
  }
}
```

**Comment by sonicdoe** [comment]
Thank you for the quick fix 👍 With Detox v16.5.1, I was able to migrate from `mocha.opts` to `.mocharc.js`.
